contracts/LockDealBundleProvider/LockDealBundleProvider.sol

```diff
@@ -102,7 +102,31 @@ contract LockDealBundleProvider is
         uint256 oldPoolId,
         uint256 newPoolId,
         uint256 splitAmount
+    ) public override onlyProvider {
+    }
+
+    function split(
+        uint256 bundlePoolId,
+        address token,
+        uint256[] memory splitAmounts,
```

Comment:
I'd like to propose a more efficient method for splitting our pools. We should maintain our base format, but introduce a ratio-based division strategy. Let's explore an example:

Consider a scenario where we have 10 pools, each holding 5 tokens. Our total pool thus sums up to 50 tokens. Now, if we want to split these pools, the maximum split amount could be up to 50. We can calculate the split ratio as total/split amount.

Subsequently, we can determine the number of splits required for each pool. Starting with the first poolId for the split, we can create new pools based on the ratio. For instance, using our example of 10 pools, there would be a creation of 11 new pools. This includes 1 pool with the first id, 9 pools created in the loop, and finally, a bundle pool with the first pool id from the input.

This strategy optimizes our splitting mechanism and offers a more balanced distribution. Your insights and suggestions are appreciated as we strive to improve our processes.

some example:
https://docs.google.com/spreadsheets/d/10ShT27mq03IgOf3OgMV5IazklqAQtklE-6Bz76kGdeM/edit#gid=2120501287